### ISSUE TITLE
The instruments are functional but remain dark with MSFS 2024.

### ISSUE THREAD

**Issue by Jbtmike** [scott-vincent/instrument-panel#63]
Hello,
First of all! You've produced an excellent tool! Thank you very much!

I tested it with MSFS 2020 and it works like a charm.

I also tested with MSFS 2024 but there is a problem with the display: the instruments work fine but they remain dark as if the  batteries were switched off.

I don't know if you've validated compatibility with MSFS 2024, but thanks in advance for anything you can do!

![IMG20241229172457](https://github.com/user-attachments/assets/8276b13c-fa17-48fb-b794-3cf7c3d17c2d)

**Comment by scott-vincent** [comment]
I don't plan on getting MSFS 2024 yet as it's too buggy atm. Maybe in 6 months time and then I will make my software compatible.

**Comment by Jbtmike** [comment]
Hello. Duly noted. Thank you for your time and your panel !

**Comment by Jbtmike** [comment]
Hello scott-vincent,
Finally, this morning I had a quick look at your code and the variables in FS2020 and FS2024.
It seems that MS has changed the names of some of them, especially in the electrical part. That's why your dcVolts and batteryLoad variables always return 0.
In instrument-panel.cpp, I changed the test at the beginning of "updateCommon()", leaving only "globals.electrics = globals.connected", recompiled, and it's usable with FS2024: the screen doesn't darken anymore.
The ammeter needle doesn't move anymore and the annunciator doesn't seem to work anymore.
I only fly single engine airplanes and haven't done any extensive tests. 
For users who want to test FS2024, it might be possible to wait for a more complete revision of your tool (instrument-data-link and instrument-panel) ;-)
I wish you a happy new year 2025 !
![msfs2024mod](https://github.com/user-attachments/assets/c6b99157-5dd0-43f3-a848-117967d52ab7)
![msfs2024mod_elec](https://github.com/user-attachments/assets/f881c6a4-8179-4b4d-885a-6a94451aa680)

**Comment by Jbtmike** [comment]
I made a mistake with visual studio, please ignore my pull request. My apologies for the inconvenience...

**Comment by scott-vincent** [comment]
The whole thing will need to be compiled against the FS2024 SDK instead of the FS2020 one so once I buy FS2024 I will probably end up creating a whole new repository and archive this one so it can continue to be used with FS2020.

Could be quite a lot of work by the looks of it. Still waiting for FS2024 to be in a better state and where the SDK is more stable otherwise I will be constantly chasing the SDK updates.

**Comment by Jbtmike** [comment]
I agree with you. I got FS2024 as a gift a week ago and I realise that there are still quite a few bugs!  I can't imagine the SDK is any better!
All the best!

**Comment by Tobair424** [comment]
I'll add that planes that migrated from msfs2020 to msfs2024 don't have this issue and the panel lights up, and also works with planes that do not have G1000 from msfs2024 in some cases

**Comment by Tobair424** [comment]
This one, for example it's a cessna 182 rg that was originally from msfs2020 ![IMG-20250106-WA0047](https://github.com/user-attachments/assets/d0301f13-0ecd-4d07-9a1c-22e6885919f8)

**Comment by Jbtmike** [comment]
HI Tobair424, Thank you for this information. I mainly did my tests with the C152 and DR400 from FS 2024. Even without considering the point related to "Instrument panel", I notice that these two planes have several bugs (including the display of the battery charge...). I'm going to import the same planes from FS 2020 to compare behaviors.
